core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainRequest.java

```diff
@@ -84,6 +90,8 @@ public ActionRequestValidationException validate() {
             }
             if (this.primary == null) {
                 validationException = addValidationError("primary must be specified", validationException);
+            } else if (this.primary && this.currentNodeId != null) {
+                validationException = addValidationError("currentNodeId can only be specified for a replica shard", validationException);
```

Comment:
we can still verify that that node has the primary and fail if that's not the case.